hw/ip/flash_ctrl/data/flash_ctrl_testplan.hjson

```diff
@@ -24,199 +24,356 @@
-            1. Read / program / erase a protected chunk
-            2. Set num words to 0 or out of bounds on read or program
-            3. Inject double bit errors when ECC is enabled
-            4. Program or read out-of-bounds info partition
-            5. Erase whole bank with bank erase not enabled
-            6. Randomly wiggle scramble enable in the middle of an op and ensure that the FSM does
-            not get locked up
+            Perform accesses in order to provoke life cycle management interface error. Related
+            covergroup is error_cg.
             '''
       milestone: V2
       tests: []
     }
     {
-      name: flash_macro_timing
+      name: error_storage
       desc: '''
-            When verifying against the real flash macro (which will be in a closed source repo),
-            check that the read / program / erase timing parameters are correct.
+            Perform accesses in order to provoke shadow register storage fault error. Related
+            covergroup is error_cg.
             '''
       milestone: V2
       tests: []
     }
     {
-      name: stress_all
+      name: secret_partition
       desc: '''
-            This enables all of the above features to fully stress all the moving parts in the
-            design.
+            Verify values of secret information partitions. Enabling by life cycle and otp is
+            required. Pages are read upon flash controller initialization.
             '''
       milestone: V2
-      tests: ["flash_ctrl_stress_all"]
+      tests: []
     }
     {
-      name: flash_init
+      name: isolation_partition
       desc: '''
-            TBD - if supported - verify flash initialization routine upon reset.
+            Verify values of isolated information partitions. Accessablity is controlled by life
+            cycle and otp.
             '''
       milestone: V2
       tests: []
     }
     {
-      name: tlul_to_vendor
+      name: interrupts
       desc: '''
-            TBD - if supported - verify TLUL split to the vendor wrapper.
+            Perform accesses in order to raise all interrupts given in register map.
+            Check behaviour of Interrupt Enable and Status Registers.
             '''
       milestone: V2
       tests: []
     }
+    {
+      name: ecc
+      desc: '''
+            Randomly enable ECC for a randomly selected set of pages. Randomly corrupt a single bit
+            in the memory that is about to be read and ensure that the ECC works - the corrupted
+            bit should be fixed. Corrupt randomly either the data or the ECC bits. Randomly corrupt
+            2 bits in the same word and ensure that the read results in error. Ensure that pages 
+            with ECC not enabled reads back corrupted data without any errors. Verify both types,
+            pre-scramble ECC(integrity ESS, 4-bits) and post-scramble ECC(reliability ECC, 8-bits).
```

Comment:
one last note on ECC, there are status registers indicating where the ECC err'd and how many ecc errors there are, as well as separate control on disabling the ecc error altogether.

It would be good to test those.  The last ecc error enable is a little bit useless, so I am thinking about removing it altogether.

Reply:
Added "Test status and control ECC bits."